Cython/Compiler/Options.py

```diff
@@ -385,6 +386,7 @@ class DEFER_ANALYSIS_OF_ARGUMENTS:
     'test_assert_c_code_has' : ('module',),
     'test_fail_if_c_code_has' : ('module',),
     'freelist': ('cclass',),
+    'formal_grammar': ('module',),
```

Comment:
I'll just flag these here as being a potentially (slightly) breaking change - these would all have been accepted at any level before although they'd only have had an effect at module level.

Reply:
Understood. I'd rather try to avoid breaking changes for 3.0.5 since we already had a few problems with odd release numbers this time. Let's wait another point release to see if we can't avoid the breakage (however small it is, I agree).

Reply:
I think what I'll do is: merge into 3.1 now, and then backport the changes to Shadow.py (but nothing else) into 3.0.x